src/domain/VendingMachine.ts

```diff
@@ -41,90 +52,126 @@ class VendingMachine implements IVendingMachine {
-      throw new Error(ERROR_MESSAGE.INCORRECT_UNIT_PRODUCT_PRICE);
-    }
-  }
-
-  validateChange(inputMoney: number, currentChange: number) {
-    if (changeValidator.isOverMax(inputMoney, currentChange)) {
-      throw new Error(ERROR_MESSAGE.OVER_AMOUNT);
-    }
-
-    if (changeValidator.isIncorrectUnit(inputMoney)) {
-      throw new Error(ERROR_MESSAGE.INCORRECT_UNIT_CHARGE_MONEY);
-    }
-  }
-
   addProduct(product: Product) {
     try {
-      this.validateProduct(product, this.products);
+      validateProduct(product, this.products);
       const newProduct = new Product(product);
 
       this.products.push(newProduct);
       storage.setLocalStorage('products', this.products);
       this.dispatch(ELEMENT_KEY.PRODUCT, 'add', newProduct);
+      this.dispatch(ELEMENT_KEY.PURCHASE, 'add', newProduct);
     } catch (error) {
-      alert(error.message);
+      showSnackBar(error.message);
     }
   }
 
   updateProduct({ targetName, name, price, quantity }) {
     try {
-      this.validateUpdateProduct(targetName, name, price, this.products);
+      validateUpdateProduct(targetName, name, price, this.products);
       const currentProduct = this.products.find((product) => product.name === targetName);
 
       currentProduct.update({ name, price, quantity } as Product);
       storage.setLocalStorage('products', this.products);
+
       this.dispatch(ELEMENT_KEY.PRODUCT, 'update', currentProduct);
+      this.dispatch(ELEMENT_KEY.PURCHASE, 'update', currentProduct);
     } catch (error) {
-      alert(error.message);
+      showSnackBar(error.message);
     }
   }
 
   deleteProduct(targetName: string) {
     const targetProduct = this.products.find((product) => product.name === targetName);
 
     this.dispatch(ELEMENT_KEY.PRODUCT, 'delete', targetProduct);
+    this.dispatch(ELEMENT_KEY.PURCHASE, 'delete', targetProduct);
     this.products = this.products.filter((product) => product.name !== targetName);
     storage.setLocalStorage('products', this.products);
   }
 
   charge(inputMoney: number) {
     try {
-      this.validateChange(inputMoney, this.amount.getAmount());
+      validateChange(inputMoney, this.amount.getAmount());
 
-      this.amount.genarateRandomCoin(inputMoney);
+      this.amount.generateRandomCoin(inputMoney);
       storage.setLocalStorage('amount', this.amount);
-      this.dispatch(ELEMENT_KEY.CHARGE, 'update');
+      this.dispatch(ELEMENT_KEY.CHARGE, 'update', this.amount);
+    } catch (error) {
+      showSnackBar(error.message);
+    }
+  }
+
+  inputMoney(money: number) {
+    try {
+      validateInputMoney(money, this.moneyInput.getAmount());
+
+      this.moneyInput.addMoney(money);
+      const userMoney = this.moneyInput.getAmount();
+      storage.setLocalStorage('userMoney', userMoney);
+      this.dispatch(ELEMENT_KEY.PURCHASE, 'input', userMoney);
+    } catch (error) {
+      showSnackBar(error.message);
+    }
+  }
+
+  purchaseProduct(targetName: string) {
```

Comment:
```suggestion
  purchaseProduct(productName: string) {
```
targetProductName, productName 등으로 네이밍하면 상품에 대한 데이터라는걸 인지하기 더 편할거같아요.

Reply:
반영 링크입니다! [e976742](https://github.com/woowacourse/javascript-vendingmachine/pull/56/commits/e976742d014fb2b783c340a425c5f33645b54ffe)